plugin/resolvergen/resolver_test.go

```diff
@@ -0,0 +1,163 @@
+package resolvergen
+
+import (
+	"fmt"
+	"go/types"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strings"
+	"testing"
+	"unicode"
+
+	"github.com/99designs/gqlgen/codegen"
+	"github.com/99designs/gqlgen/codegen/config"
+	"github.com/vektah/gqlparser/ast"
+)
+
+func TestPlugin_Name(t *testing.T) {
+	t.Run("test plugin name", func(t *testing.T) {
+		m := &Plugin{}
+		if got, want := m.Name(), "resovlergen"; got != want {
+			t.Errorf("Plugin.Name() = %v, want %v", got, want)
+		}
+	})
+}
+
+// Types for testing code generation, both Mutation and
+// MutationResolver must implement types.Type.
+type Mutation struct{}
+
+func (m *Mutation) Underlying() types.Type {
+	return m
+}
+
+func (m *Mutation) String() string {
+	return "Mutation"
+}
+
+type MutationResolver struct{}
+
+func (m *MutationResolver) Underlying() types.Type {
+	return m
+}
+
+func (m *MutationResolver) String() string {
+	return "MutationResolver"
+}
+
+func TestPlugin_GenerateCode(t *testing.T) {
+	makeData := func(cfg config.PackageConfig) *codegen.Data {
+		m := &Mutation{}
+		obj := &codegen.Object{
+			Definition: &ast.Definition{
+				Name: fmt.Sprint(m),
+			},
+			Root: true,
+			Fields: []*codegen.Field{
+				&codegen.Field{
+					IsResolver:  true,
+					GoFieldName: "Name",
+					TypeReference: &config.TypeReference{
+						GO: m,
+					},
+				},
+			},
+			ResolverInterface: &MutationResolver{},
+		}
+		obj.Fields[0].Object = obj
+		return &codegen.Data{
+			Config: &config.Config{
+				Resolver: cfg,
+			},
+			Objects: codegen.Objects{obj},
+		}
+	}
+
+	t.Run("renders expected contents", func(t *testing.T) {
+		m := &Plugin{}
```

Comment:
Maybe an easier and simpler way to write this test would be to use golden files (https://medium.com/@jarifibrahim/golden-files-why-you-should-use-them-47087ec994bf)

Reply:
We already do this using code review of the committed generated files. No need for update, just commit the generated code and push.

All this test should need to do is call the generator with some config, [eg](https://github.com/99designs/gqlgen/blob/master/plugin/modelgen/models_test.go). It may not look like its asserting anything, but really its output is tested in three ways:
 - firstly we run goimports when saving go files, so running generate should error out in the test if it generates invalid code.
 - if that fails the linter should see invalid code, and fail the branch.
 - code review catches changes. diffs are readable in github.